packages/scalecube-microservice/src/Dispatcher/index.ts

```diff
@@ -0,0 +1,57 @@
+import { ServiceCallRequest } from '../api/Dispatcher';
+import { from, iif, of } from 'rxjs6';
+import { switchMap, tap, map, mergeMap, catchError } from 'rxjs6/operators';
+
+export const createDispatcher = ({ router, serviceRegistry, getPreRequest$, postResponse$ }) => {
+  // TODO add ServiceCallResponse - serviceCall implementation
+  return ({ message, type }: ServiceCallRequest) => {
+    if (!message) {
+      throw Error('Error: data was not provided');
+    }
+
+    const routerInstance = router.route({ serviceRegistry, request: message });
+    const serviceInstance = routerInstance.service;
+    const method = serviceInstance[message.methodName];
+
+    const enrichMsgDataPreRequest = (msg) =>
+      iif(
+        () => typeof getPreRequest$ === 'function',
+        of(msg).pipe(mergeMap((msg) => getPreRequest$(of(msg)))),
+        of(msg)
+      );
+
+    const enrichMsgDataPostResponse = (msg) =>
+      iif(() => typeof postResponse$ === 'function', of(msg).pipe(mergeMap((msg) => postResponse$(of(msg)))), of(msg));
+
+    const isAsyncLoader = (service) => false; // TODO asyncService
+
+    const chain$ = enrichMsgDataPreRequest(message).pipe(
+      catchError((err) => {
+        console.warn(new Error(`dispatcher error: ${err}`));
+        return of({});
```

Comment:
you can return EMPTY - it's a special stream that instantly completes (need to be imported from 'rxjs' module)